File: Scripts/generated_codes/python/codechef_responses/6.py
'''
'''

# SOLUTION

for _ in range(int(input())):
    s1 = input()
    s2 = input()
    max_count = 0
    min_count = 0
    for i in range(len(s1)):
        if s1[i] == '?' or s2[i] == '?':
            max_count += 1
        elif s1[i] != s2[i]:
            max_count += 1
            min_count += 1
    print(min_count, max_count)

File: Scripts/generated_codes/java/6.java
"""
"""



import sys

T = int(input())

for i in range(T):
    S1 = list(input())
    S2 = list(input())
    min_diff = 0
    max_diff = 0
    for i in range(len(S1)):
        if (S1[i] == '?' and S2[i] == '?'):
            max_diff += 1
        elif (S1[i] == '?' or S2[i] == '?'):
            max_diff += 1
            min_diff += 1
        elif (S1[i] != S2[i]):
            min_diff += 1
            max_diff += 1
    print(min_diff, max_diff)